### PR TITLE
Dashboard Favorite API and UI integration

### LINKED ISSUE #407: [UI] Make frontend changes for markinga dashboard as a favorite
# Description
Currently we have a star icon to mark the dashboard as a favorite, but that marking is not getting saved. 
If the user marks the dashboard favorite, then you need to update the dashboardDetails with this info by calling the `/dashboards/update` endpoint.
 Also, if the user have any dashboards marked as favorite they should listed at the top of the list.

### PR DESCRIPTION
**Pull request by venkateshamatam**
# Description
Integrate dashboard favorite API with UI and create a new API to list favorite dashboards and pre-fill the star icon on page load according to its current favorite status

Fixes #407 

# Testing
Tested the code working for both default and non-default dashboards along with duplicate dashboards.
Tested this code by verifying the status on initial load, reload and after toggling the favorite status for a few dashboards.
Tested this code by making changes to the favorite status through postman and reloading the UI. 

# Checklist:
Before marking your pull request as ready for review, complete the following.

- [x] I have self-reviewed this PR.
- [x] I have removed all print-debugging and commented-out code that should not be merged.
- [x] I have added sufficient comments in my code, particularly in hard-to-understand areas.
- [x] I have formatted the code, if applicable. For Go, I have run `goimports -w .`.

### CODE REVIEW

**Review by nbhavana — COMMENTED** [review]
lgtm. While you are in the same code base can you list all the favorite dashboard at the top and then the rest of the dashboards. Also add unit tests for the for the new code.

File: pkg/dashboards/dashboards.go [hunk 1]
```diff
@@ -305,7 +305,12 @@ func createDashboard(dname string, orgid uint64) (map[string]string, error) {
 }
 func toggleFavorite(id string) (bool, error) {
 	// Load the dashboard JSON file
-	dashboardDetailsFname := config.GetDataPath() + "querynodes/" + config.GetHostID() + "/dashboards/details/" + id + ".json"
+	var dashboardDetailsFname string
+	if id == "10329b95-47a8-48df-8b1d-0a0a01ec6c42" || id == "a28f485c-4747-4024-bb6b-d230f101f852" || id == "bd74f11e-26c8-4827-bf65-c0b464e1f2a4" || id == "53cb3dde-fd78-4253-808c-18e4077ef0f1" {
```
Comment (AndrewHess) [inline pkg/dashboards/dashboards.go:309]: Can you make a function like `dashboardIsDefault(id)` that does this check, and then call that function here and on line 623 where you're doing a similar check? That way there's fewer opportunities for mistakes when checking if a dashboard is a default dashboard.

File: static/js/dashboards-home.js [hunk 2]
```diff
@@ -528,6 +560,9 @@ $(document).ready(async function () {
 	let normalDBs = await getAllDashboards();
 	let allDefaultDBs = await getAllDefaultDashboards();
 	let allDBs = {...normalDBs, ...allDefaultDBs}
+	let favoriteDBs = await getAllFavoriteDashboards();
+    // Convert the array of favorite dashboards to a Set for faster lookup
+	favoriteDBsSet = new Set(Object.keys(favoriteDBs));
```
Comment (AndrewHess) [inline static/js/dashboards-home.js:565]: nit: the indentation of this comment doesn't match the other code

**Review by AndrewHess — COMMENTED** [review]
Looks good, just a few minor comments

**Pull request merged**